packages/analytics/analytics-chart/src/utils/commonOptions.ts

```diff
@@ -24,21 +22,54 @@ export const tooltipBehavior = (tooltipData: TooltipState, context: ExternalTool
     const colors = tooltip.labelColors
     const valueAxis = context.chart.config?.options?.indexAxis === 'y' ? 'x' : 'y'
 
-    const isBarChart = ['horizontal_bar', 'vertical_bar'].includes(tooltipData.chartType)
+    tooltipData.tooltipContext = formatByGranularity(new Date(tooltip.dataPoints[0].parsed.x), granularity, true)
```

Comment:
Just moving around some old code. 

lineChartTooltipBehaviour needs to know about granularity while bar charts do not